### PR TITLE
Feat/bucket names crawler

### PR DESCRIPTION
**Pull request by sudiptob2**
related to #228 

- [x] Updated crawler interface
- [x] Implemented storage bucket crawler.
- [x] moved configuration logic to the crawler factory for the storage bucket

Challenges

- Even though it implements the ICrawler interface, it does not fully respect the contract. (Ex, return type is different) The code works though.

### CODE REVIEW

File: src/gcp_scanner/crawler/bucket_names_crawler.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import logging
+import sys
+from typing import List, Dict, Any, Union, Tuple, TextIO
+
+from googleapiclient import discovery, errors
+
+from gcp_scanner.crawler.interface_crawler import ICrawler
+
+
+class BucketNamesCrawler(ICrawler):
```
Comment (mshudrak) [inline src/gcp_scanner/crawler/bucket_names_crawler.py:24]: Ok, I am a bit confused :) you have bucket_names_crawler and storage_buckets_crawler. I think this file was added by accident or am I missing something?

**Review by mshudrak — CHANGES_REQUESTED** [review]
Left one comment, other than that looks pretty good.

File: src/gcp_scanner/crawler/bucket_names_crawler.py (hunk 1 above)
Comment (sudiptob2) [reply src/gcp_scanner/crawler/bucket_names_crawler.py]: @mshudrak Thanks :rocket: I did not notice this one :confused: Updated now

**Review by mshudrak — APPROVED** [review]
LGTM, THX.

**Pull request merged**